rex_xai/distributions.py

```diff
@@ -53,33 +64,21 @@ def str2distribution(d: str) -> Distribution:
 
 def random_coords(d: Optional[Distribution], *args, map=None) -> int:
     """generates random coordinates given a distribution and args"""
+
+    if args[0] < 2:
```

Comment:
`args[0]` is the space/area of the box that's being split, right? It might be good to add a warning here - it's unlikely to be relevant to users, but for people reading/modifying the code, clarifying that this condition means the box is too small to split would make the code clearer.

Alternatively/additionally, it would be helpful to add a note in the docstring on the values expected in args - i.e. it's expected to be space, height, width, then dist_args if present. This would also help make the code easier to follow, without needing to refer to where random_coords() is called.